PET_Clinic/PET_BUSINESS_TRANSACTIONS.md

```diff
@@ -0,0 +1,72 @@
+
+Property | Value
+--- | ---
+Name | Pet_Clinic_API
+Description | Pet Clinic Microservice application
+
+### Creating Application Definitions
+
+Once a business context has been created, it is opened in the editor. Now, we have to specify the application definition. In order to correlate data to an application **application mappings** are used.
+
+Create a new **application mapping** by clicking on the ![Add](../images/add_obj.gif?raw=true) plus icon in the upper right corner of the view.
+
+We know that each microservice pf the Petclinic application is running on different ports. Therefore, a **HTTP Server Port Matching** should be created which matches all requests on a specific port to an application. The Pet_Clinic_API is running on port 8080. Specify this port and save the configuration. Now, all data matching this application definition is correlated to it. You can check this by navigating to the *Data Explorer View* of the api-gateway agent. 
+
+### Creating Business Transaction Definitions
+
+In the next step, we can group the data of an application and combine them with a certain business context, for example: the business transaction *Owner* contains all data belonging to the process of editing the details of an owner.
+
+#### 1. Creating Business Transaction for Owner
+
+First, switch to the **Business Transaction Definitions** tab (at the bottom of the view) and click on the **Create** button. As before, a dialog appears where we can name the newly created business transaction definition. Enter a name, e.g. *Owner* and click on the *Finish* button.
+
+Like a application definition, a business transaction definition consists of **business transaction mappings** which are used to correlate data to a business transaction.
+
+As before, create an **URI Matcher** which accepts all data where the URI starts with the value ```/api/customer/owners```. Press ```CTRL + S``` to save the changes. And that's it.
+
+#### 2. Creating Business Transaction for Static Resources
+
+Second, we would like to see all requests that are related to static resources. Create a new business tracsaction with the name "Static Resources". As before, create an **URI Matcher** which accepts all data where the URI matches the regex ```.*(js|css|woff|woff2|png)```.  
+
+#### 3. Dynamic Business Transaction Extraction
+
+Third, we still have requests that are not mapped to a Business Transaction. Thus, we would like to dynamically extract the name of the business transaction depending on the **URI**. As before, create a new Business Trancation named *Dynamic BT* and add a **URI Matcher** which accepts all data where the URI starts with the value ```/```. Enable the "Dynamic Name Extraction" on the right side. Activate the checkbox "Extract name dynamically" and select the source based on which the name should be extracted. Choose **HTTP URI** from the dropdown list. In the **Name Pattern Specification** select **Specify name pattern** and add:
```

Comment:
Need to described that dynamic extraction is there to allow easier specification of the BTs, so that you don't have to do it manually for each BT..